paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -128,15 +128,12 @@ internal class StripeCustomerAdapter @Inject constructor(
         return getCustomerEphemeralKey()
             .mapCatching { customerEphemeralKey ->
                 setupIntentClientSecretProvider.provideSetupIntentClientSecret(customerEphemeralKey.customerId)
-            }.fold(
-                onSuccess = { it },
-                onFailure = { cause, displayMessage ->
-                    CustomerAdapter.Result.failure(
-                        cause = cause,
-                        displayMessage = displayMessage,
-                    )
-                }
-            )
+            }.getOrElse { cause, displayMessage ->
+                CustomerAdapter.Result.failure(
```

Comment:
Still seems weird that we need to map this. I'm guessing the generic types of result are different between the two.

Reply:
It's almost like we want the mapCatching to also map the failure (which all it really needs to do is cast, not create a new instance)